cms/src/main/java/uk/nhs/digital/externalstorage/s3/S3ConnectorImpl.java

```diff
@@ -54,9 +90,44 @@ public S3ObjectMetadata uploadFile(InputStream fileStream, String fileName, Stri
         return new S3ObjectMetadataImpl(resultMetadata, bucketName, objectKey);
     }
 
+    public List<PartETag> uploadParts(InputStream fileStream, final String bucketName, final String objectKey, final String uploadId) throws IOException {
+        final byte[] buffer = new byte[BUFFER_SIZE];
+        final List<PartETag> partETags = new ArrayList<>();
+        int lastBytesReadCount = 0;
+        int processedBytesCount = 0;
+        int currentPartNumber = 0;
+
+        while (lastBytesReadCount >= 0) {
+            lastBytesReadCount = fileStream.read(buffer, 0, BUFFER_SIZE);
+
+            if (lastBytesReadCount >= 0) {
+                processedBytesCount += lastBytesReadCount;
+                currentPartNumber++;
+
+                try (ByteArrayInputStream partStream = new ByteArrayInputStream(buffer, 0, lastBytesReadCount)) {
+                    UploadPartRequest req = new UploadPartRequest()
+                        .withBucketName(bucketName)
+                        .withKey(objectKey)
+                        .withUploadId(uploadId)
+                        .withInputStream(partStream)
+                        .withPartNumber(currentPartNumber)
+                        .withPartSize(lastBytesReadCount);
+
+                    partETags.add(s3.uploadPart(req).getPartETag());
```

Comment:
It appears that the unit test for `S3Connector` does not verify that the upload of individual chunks actually takes place (the `s3.upload` call).